libunwind/src/UnwindRegistersSave.S

```diff
@@ -824,6 +824,13 @@ DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
 
 #elif defined(__aarch64__)
 
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+DEFINE_LIBUNWIND_FUNCTION(__rtld_unw_getcontext)
```

Comment:
Explicitly hooking into rtld from libunwind feels pretty gross...

Reply:
I did the same for libc and libthr. Is libunwind any different?

Hooking into rtld seems to be the only way to access Executive mode registers from libunwind...

Reply:
Yes, because they’re developed by and for FreeBSD, rather than being a third-party library, and they already have such hooks.